cmd/worker/main.go

```diff
@@ -6,45 +6,63 @@ import (
 	"os"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/phuslu/log"
 
 	"github.com/hrand1005/ZoneScaler/common"
 	"github.com/hrand1005/ZoneScaler/worker"
 )
 
+const (
+	maxRetries    = 30
+	retryInterval = 1 * time.Second
+)
+
 func main() {
 	if len(os.Args) < 2 {
 		log.Fatal().Msg("Usage: ./coordinator <JSON config>")
+		return
 	}
 
 	conf, err := worker.LoadConfig(os.Args[1])
 	if err != nil {
 		log.Fatal().Err(err).Msgf("Failed to load config")
+		return
 	}
 
 	node := common.GameNode{
-		ID:            "unique-node-id",
+		ID:            uuid.NewString(),
```

Comment:
Do you mean during registration with the coordinator?

Reply:
I was looking into this, which is why I kept it open so far. I think we should discuss together how we want to do it, which is either let the coordinator define the id and return it, or we pass an ID and it can get errored from the coordinator that its already registered